parsl/app/python.py

```diff
@@ -45,6 +45,29 @@ def __call__(self, *args, **kwargs):
         else:
             dfk = self.data_flow_kernel
 
+        walltime = self.kwargs.get('walltime')
+        if walltime is not None:
+            def timeout(f, walltime):
```

Comment:
I think this code would be more readable if `def timeout` was not inline but was a separate toplevel function -  I don't think it takes in anything from scope in which it is defined in.

Reply:
will do